db/migrate/20170804161245_add_slug_to_tutorials.rb

```diff
@@ -0,0 +1,9 @@
+class AddSlugToTutorials < ActiveRecord::Migration[5.1]
+  def change
+    add_column :tutorials, :slug, :text
+
+    Tutorial.find_each(&:save!)
+
+    change_column_null :tutorials, :slug, false
+  end
```

Comment:
The end goal of this migration is to create the `slug` field and set it to `null: false`. Since we've got data in our database already, simply running the migration with `add_column :tutorials, :slug, :text, null: false` would blow up.

So instead, we first add the column then save each record which triggers FriendlyId and generates slugs for each tutorial. Lastly, the migration changes the `slug` column, adding the `null: false` constraint.

![tmyk](http://i3.kym-cdn.com/photos/images/newsfeed/000/620/434/f9e.gif)